activestorage/test/service/disk_service_test.rb

```diff
@@ -12,6 +12,22 @@ class ActiveStorage::Service::DiskServiceTest < ActiveSupport::TestCase
     assert_equal :tmp, @service.name
   end
 
+  test "url_for_direct_upload" do
+    original_url_options = Rails.application.routes.default_url_options.dup
+    Rails.application.routes.default_url_options.merge!(protocol: "http", host: "test.example.com", port: 3001)
+
+    key      = SecureRandom.base58(24)
+    data     = "Something else entirely!"
+    checksum = Digest::MD5.base64digest(data)
+
+    begin
+      assert_match(/^https:\/\/example.com\/rails\/active_storage\/disk\/.*$/,
```

Comment:
this test was inspired in the test below, what I don't understand is why the port isn't being preserved in both tests 🤔